src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2Connection.cs

```diff
@@ -480,6 +461,108 @@ private void ProcessGoAwayFrame(FrameHeader frameHeader)
             _incomingBuffer.Discard(frameHeader.Length);
         }
 
+        private async void SendSettingsAck()
+        {
+            await _writerLock.WaitAsync().ConfigureAwait(false);
+            try
+            {
+                WriteFrameHeader(new FrameHeader(0, FrameType.Settings, FrameFlags.Ack, 0));
+
+                await FlushOutgoingBytesAsync().ConfigureAwait(false);
+            }
+            finally
+            {
+                _writerLock.Release();
+            }
+        }
+
+        private async void SendPingAck(ReadOnlyMemory<byte> pingContent)
+        {
+            Debug.Assert(pingContent.Length == FrameHeader.PingLength);
+
+            await _writerLock.WaitAsync().ConfigureAwait(false);
+            try
+            {
+                WriteFrameHeader(new FrameHeader(FrameHeader.PingLength, FrameType.Ping, FrameFlags.Ack, 0));
+                _outgoingBuffer.EnsureAvailableSpace(FrameHeader.PingLength);
+                pingContent.CopyTo(_outgoingBuffer.AvailableMemory);
+                _outgoingBuffer.Commit(FrameHeader.PingLength);
+
+                await FlushOutgoingBytesAsync().ConfigureAwait(false);
+            }
+            finally
+            {
+                _writerLock.Release();
+            }
+        }
+
+        private async void SendRstStream(int streamId, Http2ProtocolErrorCode errorCode)
+        {
+            await _writerLock.WaitAsync().ConfigureAwait(false);
+            try
+            {
+                WriteFrameHeader(new FrameHeader(FrameHeader.RstStreamLength, FrameType.RstStream, FrameFlags.None, streamId));
+
+                _outgoingBuffer.AvailableSpan[0] = (byte)(((int)errorCode & 0xFF000000) >> 24);
```

Comment:
Is it safe to skip the call to `_outgoingBuffer.EnsureAvailableSpace` here?

Reply:
No, it's not. Good catch, will fix.